api/dataset.go

```diff
@@ -1070,6 +1076,73 @@ func (d *PublishCheck) Check(handle func(http.ResponseWriter, *http.Request)) ht
 
 		if versionDoc != nil {
 			if versionDoc.State == models.PublishedState {
+				defer func() {
+					if err := r.Body.Close(); err != nil {
+						log.ErrorC("could not close response body", err, nil)
+					}
+				}()
+
+				versionDoc, err := models.CreateVersion(r.Body)
```

Comment:
It is clearer to use a different name here to that of the current version also known as `versionDoc`, or change the one above to `currentVersion`?